packages/swing-store/src/swingStore.js

```
@@ -907,34 +929,37 @@ export async function importSwingStore(exporter, dirPath = null, options = {}) {
+    const [tag] = key.split('.', 1);
+    const subKey = key.substring(tag.length + 1);
+    if (tag === 'kv') {
+      // 'kv' keys contain individual kvStore entries
+      if (value == null) {
+        // Note '==' rather than '===': any nullish value implies deletion
+        kernelStorage.kvStore.delete(subKey);
+      } else {
+        kernelStorage.kvStore.set(subKey, value);
+      }
+    } else if (tag === 'transcript' || tag === 'snapshot') {
+      // 'transcript' and 'snapshot' keys contain artifact description info.
       assert(value); // make TypeScript shut up
-      const { type, vatID, isCurrent, pos } = parseExportKey(key);
+      const { vatID, isCurrent, pos } = parseExportKey(subKey);
       if (isCurrent) {
         const vatInfo = vatArtifacts.get(vatID) || {};
-        if (type === 'snapshot') {
+        if (tag === 'snapshot') {
           // `export.snapshot.{vatID}.current` directly identifies the current snapshot artifact
           vatInfo.snapshotKey = value;
-        } else if (type === 'transcript') {
+        } else if (tag === 'transcript') {
           // `export.transcript.${vatID}.current` contains a metadata record for the current
           // state of the current transcript span as of the time of export
           const metadata = JSON.parse(value);
-          vatInfo.transcriptKey = artifactKey(type, vatID, metadata.startPos);
+          vatInfo.transcriptKey = artifactKey(tag, vatID, metadata.startPos);
           artifactMetadata.set(vatInfo.transcriptKey, metadata);
         }
         vatArtifacts.set(vatID, vatInfo);
       } else {
-        artifactMetadata.set(artifactKey(type, vatID, pos), JSON.parse(value));
+        artifactMetadata.set(artifactKey(tag, vatID, pos), JSON.parse(value));
       }
-    } else if (value == null) {
-      // Note '==' rather than '===': any nullish value implies deletion
-      kernelStorage.kvStore.delete(key);
-    } else {
-      kernelStorage.kvStore.set(key, value);
     }
```

Let's add an assertion here so we fail if trying to import an unknown `tag`